design-docs/improved-console.md

```diff
@@ -0,0 +1,283 @@
+# Improved Console Output
+
+## Primary Goals
+* Improve perceived performance by making user aware of background work
+* Modern feel
+* User extensibility
+
+### User Goals for Console
+In priority order, the user wants to answer the following questions:
+
+* When will my build finish?
+    * Soon (should I keep watching it) or later ("I'll check back")
+* Is there anything I should know about my build right now? (e.g. tests have failed, warnings)
+* What is Gradle doing right now?
+* Is my build successful?
+* What command did I execute?
+
+## Story: Display parallel work in-progress independently by operation
+Show incomplete ProgressOperations on separate lines, up to a specified maximum number. 
+
+### Implementation
+A `RichConsole` has 
+ - an appendable `TextArea` (mainArea)
+ - a `ProgressLabel` (consisting of a `ProgressBar` and `StatusText`)
+ - a fixed-height, random-access `TextArea` (progressArea)
+ 
+The mainArea is displayed at the top for rendering important messages like warnings
+and errors.
+
+`--quiet`, `--info`, and `--debug` log level flags affect the output visible in the main TextArea,
+However, the default will change so that `LIFECYCLE` logs are no longer displayed.
```

Comment:
I would think about including some context when we log an important message, as often the warning/error messages don't include enough context for a human to know which build operation (such as a task) the message relates to. This is particularly important when running in parallel, and logging things like compile error messages.

Currently the lifecycle logging serves this purpose (not perfectly), so if we remove it, we should add something else to help the user make sense of the error and logging messages.